test/SlowTests/Server/Documents/DataArchival/DataArchivalIndexingTests.cs

```diff
@@ -0,0 +1,483 @@
+                };
+
+            Reduce = results => from result in results
+                group result by result.Name into g
+                select new Result
+                {
+                    Name = g.Key,
+                    Count = g.Sum(x => x.Count)
+                };
+        }
+    }
+    
+    [RavenTheory(RavenTestCategory.Indexes)]
+    [RavenData(SearchEngineMode = RavenSearchEngineMode.All)]
+    public async Task CanArchiveEntity_WillNotBeRelevantMapReduceIndex(Options options)
+    {
+        using (var store = GetDocumentStore(options))
+        {
+            
+            // Spin up the index
+            await new Companies_NamesCount().ExecuteAsync(store);
+            
+            // Insert document with archive time before activating the archival
+            var company = new Company {Name = "Company Name"};
+            var retires = SystemTime.UtcNow.AddMinutes(5);
+            using (var session = store.OpenAsyncSession())
+            {
+                await session.StoreAsync(company);
+                var metadata = session.Advanced.GetMetadataFor(company);
+                metadata[Constants.Documents.Metadata.ArchiveAt] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                await session.SaveChangesAsync();
+            }
+
+            // Make sure that the company is not skipped while indexing yet
+            using (var session = store.OpenAsyncSession())
+            {
+                await Indexes.WaitForIndexingAsync(store);
+                var companies = await session.Query<Companies_NamesCount.Result, Companies_NamesCount>().ToListAsync();
+                Assert.Equal(1, companies.Count);
+            }
+
+            // Activate the archival
+            await SetupDataArchival(store);
+
+            var database = await Databases.GetDocumentDatabaseInstanceFor(store);
+            database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
+            var documentsArchiver = database.DataArchivist;
+            await documentsArchiver.ArchiveDocs();
+
+            using (var session = store.OpenAsyncSession())
+            {
+                var archivedCompany = await session.LoadAsync<Company>(company.Id);
+                var metadata = session.Advanced.GetMetadataFor(archivedCompany);
+                Assert.DoesNotContain(Constants.Documents.Metadata.ArchiveAt, metadata.Keys);
+                Assert.Contains(Constants.Documents.Metadata.Collection, metadata.Keys);
+                Assert.Contains(Constants.Documents.Metadata.Archived, metadata.Keys);
+                Assert.Equal(true, metadata[Constants.Documents.Metadata.Archived]);
+                
+                // Make sure that the company is skipped while indexing (auto map index)
+                var companies = await session.Query<Companies_NamesCount.Result, Companies_NamesCount>().ToListAsync();
+                await AssertWaitForCountAsync(() => Task.FromResult(companies) , 0);
```

Comment:
This is not correct  since `companies` is already materialized (you already called `.ToListAsync();`)

Reply:
fa3d630d28cf589f44d6b786f7fbff3d46fdcb10 fixed here